src/directions/helpers.ts

```diff
@@ -202,3 +203,200 @@ export function getReverseGeocodedAddresses(
       callback(new Array(positions.length).fill(""));
     });
 }
+
+/**
+ * Creates Turf.js polygons from GeoJSON feature collection, handling both Polygon and MultiPolygon
+ *
+ * @param geojson - GeoJSON FeatureCollection containing country boundaries
+ * @returns Array of Turf.js polygon features
+ */
+//keep any for now, will typecast to geojson later
+export function createPolygons(geojson: any): TurfPolygon[] {
+  if (!geojson?.features) return [];
+
+  //keep any for now, will typecast to geojson.feature later
+  return geojson.features.reduce((polygons: TurfPolygon[], feature: any) => {
+    try {
+      if (feature.geometry.type === "Polygon") {
+        polygons.push(turf.polygon(feature.geometry.coordinates));
+      } else if (feature.geometry.type === "MultiPolygon") {
+        // For MultiPolygon, each coordinate array represents a separate polygon
+        feature.geometry.coordinates.forEach((polygonCoords: number[][][]) => {
+          polygons.push(turf.polygon(polygonCoords));
+        });
+      }
+    } catch (error) {
+      console.error("Error creating polygon:", error);
+    }
+    return polygons;
+  }, []);
+}
+
+// Initialize country polygons once
+const usaPolygons = createPolygons(usaGeoJson);
+const myanmarPolygons = createPolygons(myanmarGeoJson);
+const liberiaPolygons = createPolygons(liberiaGeoJson);
+
+/**
+ * Determines if a point is within any polygon in the given array
+ *
+ * @param point - Turf.js point feature
+ * @param polygons - Array of Turf.js polygon features
+ * @returns Boolean indicating if point is within any polygon
+ */
+export function isPointInPolygons(point: number[], polygons: TurfPolygon[]): boolean {
+  if (!point || polygons.length === 0) return false;
+  // Check all polygons for point to be in any of them
+  return polygons.some((polygon) => polygon && turf.booleanPointInPolygon(point, polygon));
+}
+
+/**
+ * Determines the appropriate unit system based on the country of the provided coordinates. Returns UnitSystem.IMPERIAL
+ * for addresses in US, Liberia, and Myanmar, and UnitSystem.METRIC for all other countries.
+ *
+ * @param options - Object containing unitSystem preference
+ * @param response - OriginsResponse or OriginResponse containing location information
+ * @returns UnitSystem.IMPERIAL if location is in USA, Myanmar, or Liberia; UnitSystem.METRIC otherwise
+ */
+// keep any for now, need to fix _convertAmazonResponseToGoogleResponse types first
```

Comment:
Can use the proper types for the `options` input. Can modify the `response` input to instead pass in a `LatLng` class.